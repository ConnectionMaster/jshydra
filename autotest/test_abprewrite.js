// Arguments: --trueast abprewrite_source.js
// Name: ABP rewriting test

include("../scripts/abprewrite.js");
