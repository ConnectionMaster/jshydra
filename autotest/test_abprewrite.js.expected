/*
 * This Source Code is subject to the terms of the Mozilla Public License
 * version 2.0 (the "License"). You can obtain a copy of the License at
 * http://mozilla.org/MPL/2.0/.
 */

//
// This file has been generated automatically from Adblock Plus for Firefox
// source code. DO NOT MODIFY, change the original source code instead.
//
// Relevant repositories:
// * https://hg.adblockplus.org/adblockplus/
// * https://hg.adblockplus.org/jshydra/
//

const Cl = Components.unknown;
var Utils = require("utils").Utils;
var _tempVar0 = require("filterClasses");
var Filter = _tempVar0.Filter;
var BlockingFilter = _tempVar0.BlockingFilter;
var foo;
var bar = 2;
var bas;
var _tempVar1 = foo();
var a = _tempVar1[0];
var b = _tempVar1[1];
var _tempVar2 = [1, 2];
a = _tempVar2[0];
b = _tempVar2[1];
var y = foo().x;
var _tempVar3 = foo();
var v1 = _tempVar3.k1;
var v2 = _tempVar3.k2;
for (var _loopIndex4 = 0; _loopIndex4 < fooList.length; ++_loopIndex4)
{
  var v = fooList[_loopIndex4];
  alert(v);
}
for (var _loopIndex5 = 0; _loopIndex5 < fooList.length; ++_loopIndex5)
{
  var _tempVar6 = fooList[_loopIndex5];
  var a = _tempVar6[0];
  var b = _tempVar6[1];
  a += b;
  b -= a;
}
for (var _loopIndex7 = 0; _loopIndex7 < fooList.length; ++_loopIndex7)
{
  var _tempVar8 = fooList[_loopIndex7];
  var a = _tempVar8[0];
  var b = _tempVar8[1];
}
for (var _loopIndex9 = 0; _loopIndex9 < fooList.length; ++_loopIndex9)
{
  k = fooList[_loopIndex9];
  alert(k);
}
var a = function()
{
  return 1;
};
var b =
{
  get foo()
  {
    return 1;
  }
};

function genFunc()
{
  var _generatorResult10 = [];
  for (var i = 0; i < 10; i++)
  {
    _generatorResult10.push(i);
  }
  return _generatorResult10;
}
var a = function()
{
  var _generatorResult10 = [];
  for (var i = 0; i < 10; i++)
  {
    _generatorResult10.push(i);
  }
  return _generatorResult10;
};
